go/common/viewingkey/viewing_key.go

```diff
@@ -103,9 +106,48 @@ func GenerateSignMessage(vkPubKey []byte) string {
 	return SignedMsgPrefix + hex.EncodeToString(vkPubKey)
 }
 
-// GenerateSignMessageOG creates the message to be signed by Obscuro Gateway (new format)
-// format is expected to be "Register <userID> for <Account>" (with the account in lowercase)
-func GenerateSignMessageOG(vkPubKey []byte, addr *gethcommon.Address) string {
-	userID := crypto.Keccak256Hash(vkPubKey).Bytes()
-	return fmt.Sprintf("Register %s for %s", hex.EncodeToString(userID), strings.ToLower(addr.Hex()))
+// GetEIP712RawData Get raw data (bytes) for an EIP-712 message used to authenticate an address with user
+func GetEIP712RawData(userID string, address *gethcommon.Address) ([]byte, error) {
+	types := apitypes.Types{
```

Comment:
can we use some go structs and let go do the json marshalling?

Reply:
I tried it, but I think this is the best approach for doing it. Same approach is used also by others.. for example: https://blog.hook.xyz/validate-eip-712/